mem backend fails backend integration test suite

running `VFS_INTEGRATION_LOCATIONS="mem:///vfs_test/" go test -count=1 -tags=vfsintegration -v ./backend/testsuite/` results in:
```go

=== RUN   TestVFS
=== RUN   TestVFS/TestScheme
************** TESTING scheme: mem **************
****** testing vfs.FileSystem ******
****** testing vfs.Location ******
****** testing vfs.File ******
** location mem:///vfs_test/dstLoc/ **
    suite.go:63: test panicked: runtime error: invalid memory address or nil pointer dereference
        goroutine 8 [running]:
        runtime/debug.Stack(0xc000189200, 0x1b89600, 0x24a04f0)
        	/Users/john.judd/.gvm/gos/go1.15.12/src/runtime/debug/stack.go:24 +0x9f
        github.com/stretchr/testify/suite.failOnPanic(0xc0003f4780)
        	/Users/john.judd/go/pkg/mod/github.com/stretchr/testify@v1.6.1/suite/suite.go:63 +0x57
        panic(0x1b89600, 0x24a04f0)
        	/Users/john.judd/.gvm/gos/go1.15.12/src/runtime/panic.go:969 +0x1b9
        github.com/c2fo/vfs/v5/backend/mem.(*File).MoveToLocation(0xc000440ac0, 0x1eac5c0, 0xc00079c8d0, 0x3, 0x3, 0x1, 0x1)
        	/Users/john.judd/go/src/github.com/c2fo/vfs/backend/mem/file.go:361 +0x622
        github.com/c2fo/vfs/v5/backend/testsuite.(*vfsTestSuite).File(0xc0003f1400, 0x1eac5c0, 0xc00046fc20)
        	/Users/john.judd/go/src/github.com/c2fo/vfs/backend/testsuite/backend_integration_test.go:796 +0x1aa8
        github.com/c2fo/vfs/v5/backend/testsuite.(*vfsTestSuite).TestScheme(0xc0003f1400)
        	/Users/john.judd/go/src/github.com/c2fo/vfs/backend/testsuite/backend_integration_test.go:119 +0x185
        reflect.Value.call(0xc0004772c0, 0xc000010f30, 0x13, 0x1cd89cf, 0x4, 0xc00008de30, 0x1, 0x1, 0xc00008dcf8, 0x100d88a, ...)
        	/Users/john.judd/.gvm/gos/go1.15.12/src/reflect/value.go:476 +0x8c7
        reflect.Value.Call(0xc0004772c0, 0xc000010f30, 0x13, 0xc00008de30, 0x1, 0x1, 0x3a, 0xc00007ce50, 0x1055e59)
        	/Users/john.judd/.gvm/gos/go1.15.12/src/reflect/value.go:337 +0xb9
        github.com/stretchr/testify/suite.Run.func1(0xc0003f4780)
        	/Users/john.judd/go/pkg/mod/github.com/stretchr/testify@v1.6.1/suite/suite.go:158 +0x379
        testing.tRunner(0xc0003f4780, 0xc000418cf0)
        	/Users/john.judd/.gvm/gos/go1.15.12/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/Users/john.judd/.gvm/gos/go1.15.12/src/testing/testing.go:1168 +0x2b3
--- FAIL: TestVFS (0.01s)
    --- FAIL: TestVFS/TestScheme (0.00s)
FAIL
FAIL	github.com/c2fo/vfs/v5/backend/testsuite	0.251s
FAIL
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 - Fixed golangci-lint issues
+- Fixed #90 - integration were tests failing on move for mem-to-mem move and bad mutex unlock call when doing mem to non-mem move
 
 ## [6.4.0]
 

--- a/backend/mem/file.go
+++ b/backend/mem/file.go
@@ -375,8 +375,8 @@ func (f *File) MoveToLocation(location vfs.Location) (vfs.File, error) {
 				return file, nil
 			}
 		}
+		f.memFile.location.FileSystem().(*FileSystem).mu.Unlock()
 	}
-	f.memFile.location.FileSystem().(*FileSystem).mu.Unlock()
 	// if the file doesn't yet exist at the location, create it there
 	newFile, err := location.NewFile(f.Name())
 	if err != nil {
@@ -447,7 +447,7 @@ func (f *File) Delete(_ ...options.DeleteOption) error {
 			thisObj.i = nil
 			thisObj = nil
 			// setting that key to nil so it truly no longer lives on this system
-			mapRef[loc.Volume()][str] = nil
+			delete(mapRef[loc.Volume()], str)
 		}
 	}
 

